projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-search.component.ts

```diff
@@ -46,17 +46,18 @@ export class IgxExcelStyleSearchComponent {
 
     public onCheckboxChange(eventArgs: IChangeCheckboxEventArgs) {
         const selectAll = this.column.grid.resourceStrings.igx_grid_excel_select_all;
-        if (eventArgs.checkbox.value.value === selectAll) {
+        const selectedIndex = this.data.indexOf(eventArgs.checkbox.value);
+        if (eventArgs.checkbox.value.value === selectAll && selectedIndex === 0) {
```

Comment:
You can remove `eventArgs.checkbox.value.value === selectAll` as `Select All` will always be the first item in the list and it duplicates the second check in the if clause.